modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryHistoryMetricsValue.java

```diff
@@ -53,6 +53,15 @@
     /** Maximum bytes written on disk in total by query. */
     private final long maxTotalBytesWrittenOnDisk;
 
+    /** Distributed joins. */
+    private final boolean distributedJoins;
```

Comment:
I think the right place for all of these flags is in QueryHistoryKey class because each of them could drastically affect resource consumption (time required as well as memory). Thus it's better to distinguish execution of the same query but with different flags

Reply:
Moved to the key class.